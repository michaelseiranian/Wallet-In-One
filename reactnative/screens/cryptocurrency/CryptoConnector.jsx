import React from "react";
import {
  TouchableOpacity,
  ScrollView,
  StyleSheet,
  Text,
  View,
  Image,
} from "react-native";
import useCryptoWallet from "../crypto_wallet/useCryptoWallet";
import {useTheme} from '../../src/theme/ThemeProvider'
import exchanges from './exchanges.json'
import {styles} from '../All_Styles.style.js';
import blockchains from "./blockchains.json";
import getCryptoIcon from "./icons/icon";

/**
 * Component that displays a list of cryptocurrency wallets and exchanges that can be added, which is defined in the
 * blockchains and exchanges json files.
 */
export default function CryptoConnector(props) {

  const {connectWallet} = useCryptoWallet();
  const {dark, colors, setScheme} = useTheme();

  const stylesInternal = StyleSheet.create({
    asset: {
      padding: 10,
      marginVertical: 5,
      borderRadius: 10,
      flexDirection: "row",
    },
    assetImage: {
      width: 30,
      height: 30,
    },
  });

  return (
    <ScrollView style={[styles(dark, colors).container, {paddingHorizontal: 20, paddingTop: 20}]}>

      <View style={{flexDirection: "row", alignItems: "center"}}>
        <View style={{flex: 1, flexDirection: "column"}}>
          <Text style={[styles(dark, colors).largeTextBold, {alignSelf: "center"}]}>Connect Wallet</Text>
        </View>
      </View>

      {
        blockchains.map((blockchain) =>

          <TouchableOpacity
            key={blockchain.symbol}
            onPress={() => props.navigation.navigate("WalletConnector",
              {connectWallet: connectWallet, cryptocurrency: blockchain.name, symbol: blockchain.symbol})}
          >
            <View style={[stylesInternal.asset, {backgroundColor: colors.primary}]}>
              <View style={{alignItems: "center", justifyContent: "center", paddingRight: 10}}>
                <Image style={stylesInternal.assetImage} source={getCryptoIcon(blockchain.symbol)}/>
              </View>
              <Text style={{fontSize: 25, fontWeight: "700", color: colors.text}}>
                {blockchain.name}
              </Text>
            </View>
          </TouchableOpacity>
        )
      }

      <View style={{flexDirection: "row", alignItems: "center"}}>
        <View style={{flex: 1, flexDirection: "column"}}>
          <Text style={[styles(dark, colors).largeTextBold, {alignSelf: "center"}]}>Connect Exchange</Text>
        </View>
      </View>

      {
        exchanges.map((exchange) =>
          <TouchableOpacity
            key={exchange.name}
            onPress={() => props.navigation.navigate('Exchange Credentials', {exchange: exchange.name})}
          >
            <View style={[stylesInternal.asset, {backgroundColor: colors.primary}]}>
              <View style={{alignItems: "center", justifyContent: "center", paddingRight: 10}}>
                <Image style={stylesInternal.assetImage} source={getCryptoIcon(exchange.name)}/>
              </View>
              <Text style={{fontSize: 25, fontWeight: "700", color: colors.text}}>
                {exchange.name}
              </Text>
            </View>
          </TouchableOpacity>)
      }
      <View style={{paddingBottom:20}}>
      </View>

    </ScrollView>
  );
}
