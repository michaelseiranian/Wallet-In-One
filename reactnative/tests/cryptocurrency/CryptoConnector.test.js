import {act, fireEvent, render, screen, waitFor} from "@testing-library/react-native";
import CryptoConnector from "../../screens/cryptocurrency/CryptoConnector";

describe('<CryptoConnector />', () => {

  const params = {
    navigation: {
      navigate: jest.fn(),
      goBack: jest.fn()
    }
  };


  it('snapshot test', async () => {
    const snapshot = render(<CryptoConnector {...params} />)

    expect(snapshot).toMatchSnapshot()

    // Test buttton
    fireEvent.press(await screen.getByText('Bitcoin'))
    fireEvent.press(await screen.getByText('Binance'))

  })

  it('wallet and exchange list test', async () => {
    const snapshot = render(<CryptoConnector {...params} />)

    // Wallets
    expect(screen.getByText('Bitcoin'))
    expect(screen.getByText('Bitcoin-Cash'))
    expect(screen.getByText('Litecoin'))
    expect(screen.getByText('Dogecoin'))
    expect(screen.getByText('Dash'))
    expect(screen.getByText('Groestlcoin'))
    expect(screen.getByText('Zcash'))
    expect(screen.getByText('eCash'))

    // Exchanges
    expect(screen.getByText('Binance'))
    expect(screen.getByText('GateIo'))
    expect(screen.getByText('CoinList'))
    expect(screen.getByText('CoinBase'))
    expect(screen.getByText('Kraken'))

  })

})