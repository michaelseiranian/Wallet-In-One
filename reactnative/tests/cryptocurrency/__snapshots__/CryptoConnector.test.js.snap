// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CryptoConnector /> snapshot test 1`] = `
<RCTScrollView
  style={
    [
      {
        "backgroundColor": "white",
        "flex": 1,
      },
      {
        "paddingHorizontal": 20,
        "paddingTop": 20,
      },
    ]
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "black",
                "fontSize": 40,
                "fontWeight": "800",
              },
              {
                "alignSelf": "center",
              },
            ]
          }
        >
          Connect Wallet
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 10,
              "flexDirection": "row",
              "marginVertical": 5,
              "padding": 10,
            },
            {
              "backgroundColor": "purple",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "700",
            }
          }
        >
          Bitcoin
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 10,
              "flexDirection": "row",
              "marginVertical": 5,
              "padding": 10,
            },
            {
              "backgroundColor": "purple",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "700",
            }
          }
        >
          Bitcoin-Cash
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 10,
              "flexDirection": "row",
              "marginVertical": 5,
              "padding": 10,
            },
            {
              "backgroundColor": "purple",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "700",
            }
          }
        >
          Litecoin
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 10,
              "flexDirection": "row",
              "marginVertical": 5,
              "padding": 10,
            },
            {
              "backgroundColor": "purple",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "700",
            }
          }
        >
          Dogecoin
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 10,
              "flexDirection": "row",
              "marginVertical": 5,
              "padding": 10,
            },
            {
              "backgroundColor": "purple",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "700",
            }
          }
        >
          Dash
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 10,
              "flexDirection": "row",
              "marginVertical": 5,
              "padding": 10,
            },
            {
              "backgroundColor": "purple",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "700",
            }
          }
        >
          Groestlcoin
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 10,
              "flexDirection": "row",
              "marginVertical": 5,
              "padding": 10,
            },
            {
              "backgroundColor": "purple",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "700",
            }
          }
        >
          Zcash
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 10,
              "flexDirection": "row",
              "marginVertical": 5,
              "padding": 10,
            },
            {
              "backgroundColor": "purple",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "700",
            }
          }
        >
          eCash
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "black",
                "fontSize": 40,
                "fontWeight": "800",
              },
              {
                "alignSelf": "center",
              },
            ]
          }
        >
          Connect Exchange
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 10,
              "flexDirection": "row",
              "marginVertical": 5,
              "padding": 10,
            },
            {
              "backgroundColor": "purple",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "700",
            }
          }
        >
          Binance
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 10,
              "flexDirection": "row",
              "marginVertical": 5,
              "padding": 10,
            },
            {
              "backgroundColor": "purple",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "700",
            }
          }
        >
          GateIo
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 10,
              "flexDirection": "row",
              "marginVertical": 5,
              "padding": 10,
            },
            {
              "backgroundColor": "purple",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "700",
            }
          }
        >
          CoinList
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 10,
              "flexDirection": "row",
              "marginVertical": 5,
              "padding": 10,
            },
            {
              "backgroundColor": "purple",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "700",
            }
          }
        >
          CoinBase
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 10,
              "flexDirection": "row",
              "marginVertical": 5,
              "padding": 10,
            },
            {
              "backgroundColor": "purple",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "700",
            }
          }
        >
          Kraken
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "paddingBottom": 20,
        }
      }
    />
  </View>
</RCTScrollView>
`;
